File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/GetCountFromServerOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class GetCountFromServerOptions {

    @NonNull
    private final String reference;

    @Nullable
    private final String databaseId;

    public GetCountFromServerOptions(@NonNull String reference, @Nullable String databaseId) {
        this.reference = reference;
        this.databaseId = databaseId;
    }

    @NonNull
    public String getReference() {
        return reference;
    }

    @Nullable
    public String getDatabaseId() {
        return databaseId;
    }
}

File: packages/firestore/ios/Plugin/Classes/Options/GetCountFromServerOptions.swift
import Foundation

@objc public class GetCountFromServerOptions: NSObject {
    private let reference: String
    private let databaseId: String?

    init(reference: String, databaseId: String? = nil) {
        self.reference = reference
        self.databaseId = databaseId
    }

    func getReference() -> String {
        return reference
    }

    func getDatabaseId() -> String? {
        return databaseId
    }
}
